lib/galaxy/webapps/galaxy/api/datatypes.py

```diff
@@ -73,6 +79,71 @@ async def index(
         """Gets the list of all available data types."""
         return view_index(self.datatypes_registry, extension_only, upload_only)
 
+    @router.get(
+        "/api/datatypes/{datatype}",
```

Comment:
I think there's a conflict here with https://github.com/galaxyproject/galaxy/pull/20190/files#diff-7d2152a780aa7c049cf34735fcc435ee1a462dbe4761856524bb87700cedbb17R148 ... move this to the bottom of the file and I think that'll solve the API issues